File: SimpleOneOs.py
import os
import shutil

class SimpleOneOS:
    def __init__(self):
        print("Welcome to SimpleOneOS! Type 'help' to see available commands.")

    def create_file(self, filename):
        with open(filename, 'w') as f:
            pass
        print(f"File '{filename}' created successfully.")

    def create_directory(self, dirname):
        os.makedirs(dirname, exist_ok=True)
        print(f"Directory '{dirname}' created successfully.")

    def list_contents(self, path='.'):
        contents = os.listdir(path)
        for item in contents:
            print(item)

    def delete_file(self, filename):
        try:
            os.remove(filename)
            print(f"File '{filename}' deleted successfully.")
        except FileNotFoundError:
            print(f"File '{filename}' not found.")

    def delete_directory(self, dirname):
        try:
            shutil.rmtree(dirname)
            print(f"Directory '{dirname}' deleted successfully.")
        except FileNotFoundError:
            print(f"Directory '{dirname}' not found.")

    def rename(self, old_name, new_name):
        try:
            os.rename(old_name, new_name)
            print(f"'{old_name}' renamed to '{new_name}'.")
        except FileNotFoundError:
            print(f"'{old_name}' not found.")

    def run(self):
        while True:
            command = input("SimpleOneOS> ").strip().split()
            if len(command) == 0:
                continue
            if command[0].lower() == 'exit':
                print("Exiting SimpleOneOS. Goodbye!")
                break
            elif command[0].lower() == 'help':
                self.show_help()
            elif command[0].lower() == 'createfile':
                if len(command) > 1:
                    self.create_file(command[1])
                else:
                    print("Usage: createfile <filename>")
            elif command[0].lower() == 'createdir':
                if len(command) > 1:
                    self.create_directory(command[1])
                else:
                    print("Usage: createdir <dirname>")
            elif command[0].lower() == 'list':
                if len(command) > 1:
                    self.list_contents(command[1])
                else:
                    self.list_contents()
            elif command[0].lower() == 'deletefile':
                if len(command) > 1:
                    self.delete_file(command[1])
                else:
                    print("Usage: deletefile <filename>")
            elif command[0].lower() == 'deletedir':
                if len(command) > 1:
                    self.delete_directory(command[1])
                else:
                    print("Usage: deletedir <dirname>")
            elif command[0].lower() == 'rename':
                if len(command) > 2:
                    self.rename(command[1], command[2])
                else:
                    print("Usage: rename <oldname> <newname>")
            else:
                print(f"Unknown command: {command[0]}")

    def show_help(self):
        print("""
        Available commands:
        createfile <filename>   - Create a new empty file
        createdir <dirname>     - Create a new directory
        list [path]             - List files and directories
        deletefile <filename>   - Delete a file
        deletedir <dirname>     - Delete a directory and its contents
        rename <oldname> <newname> - Rename a file or directory
        exit                    - Exit the OS
        """)

if __name__ == '__main__':
    os_shell = SimpleOneOS()
    os_shell.run()

File: SimpleOneOS.c
#include <iostream>
#include <filesystem>
#include <fstream>

namespace fs = std::filesystem;

class SimpleOneOS {
public:
    SimpleOneOS() {
        std::cout << "Welcome to SimpleOneOS! Type 'help' to see available commands.\n";
    }

    void createFile(const std::string& filename) {
        std::ofstream file(filename);
        if (file) {
            std::cout << "File '" << filename << "' created successfully.\n";
        } else {
            std::cout << "Failed to create file '" << filename << "'.\n";
        }
    }

    void createDirectory(const std::string& dirname) {
        if (fs::create_directories(dirname)) {
            std::cout << "Directory '" << dirname << "' created successfully.\n";
        } else {
            std::cout << "Failed to create directory '" << dirname << "' or it already exists.\n";
        }
    }

    void listContents(const std::string& path = ".") {
        for (const auto& entry : fs::directory_iterator(path)) {
            std::cout << entry.path().filename().string() << std::endl;
        }
    }

    void deleteFile(const std::string& filename) {
        if (fs::remove(filename)) {
            std::cout << "File '" << filename << "' deleted successfully.\n";
        } else {
            std::cout << "File '" << filename << "' not found.\n";
        }
    }

    void deleteDirectory(const std::string& dirname) {
        if (fs::remove_all(dirname) > 0) {
            std::cout << "Directory '" << dirname << "' deleted successfully.\n";
        } else {
            std::cout << "Directory '" << dirname << "' not found.\n";
        }
    }

    void renameFile(const std::string& oldName, const std::string& newName) {
        try {
            fs::rename(oldName, newName);
            std::cout << "'" << oldName << "' renamed to '" << newName << "'.\n";
        } catch (const fs::filesystem_error& e) {
            std::cout << "Error renaming: " << e.what() << std::endl;
        }
    }

    void run() {
        std::string command;
        while (true) {
            std::cout << "SimpleOneOS> ";
            std::getline(std::cin, command);
            std::istringstream iss(command);
            std::string cmd;
            iss >> cmd;

            if (cmd == "exit") {
                std::cout << "Exiting SimpleOneOS. Goodbye!\n";
                break;
            } else if (cmd == "help") {
                showHelp();
            } else if (cmd == "createfile") {
                std::string filename;
                iss >> filename;
                if (!filename.empty()) {
                    createFile(filename);
                } else {
                    std::cout << "Usage: createfile <filename>\n";
                }
            } else if (cmd == "createdir") {
                std::string dirname;
                iss >> dirname;
                if (!dirname.empty()) {
                    createDirectory(dirname);
                } else {
                    std::cout << "Usage: createdir <dirname>\n";
                }
            } else if (cmd == "list") {
                std::string path;
                iss >> path;
                if (!path.empty()) {
                    listContents(path);
                } else {
                    listContents();
                }
            } else if (cmd == "deletefile") {
                std::string filename;
                iss >> filename;
                if (!filename.empty()) {
                    deleteFile(filename);
                } else {
                    std::cout << "Usage: deletefile <filename>\n";
                }
            } else if (cmd == "deletedir") {
                std::string dirname;
                iss >> dirname;
                if (!dirname.empty()) {
                    deleteDirectory(dirname);
                } else {
                    std::cout << "Usage: deletedir <dirname>\n";
                }
            } else if (cmd == "rename") {
                std::string oldName, newName;
                iss >> oldName >> newName;
                if (!oldName.empty() && !newName.empty()) {
                    renameFile(oldName, newName);
                } else {
                    std::cout << "Usage: rename <oldname> <newname>\n";
                }
            } else {
                std::cout << "Unknown command: " << cmd << "\n";
            }
        }
    }

private:
    void showHelp() {
        std::cout << "Available commands:\n"
                  << "createfile <filename>   - Create a new empty file\n"
                  << "createdir <dirname>     - Create a new directory\n"
                  << "list [path]             - List files and directories\n"
                  << "deletefile <filename>   - Delete a file\n"
                  << "deletedir <dirname>     - Delete a directory and its contents\n"
                  << "rename <oldname> <newname> - Rename a file or directory\n"
                  << "exit                    - Exit the OS\n";
    }
};

int main() {
    SimpleOneOS os_shell;
    os_shell.run();
    return 0;
}
